### DISCUSSION TITLE
Check if the server response actually changed

### DISCUSSION THREAD

**Discussion by mlisthenewcool** [TanStack/query#2273] [Q&A] [answered]
With very basic architecture like this : 

```typescript
const Item = ({ _id, title }: { _id: string; title: string }) => (
  <li>
    item {_id} {title}
  </li>
)

const Items = ({ items }: { items: { _id: string; title: string }[] }) => (
  <ul>
    {items.map((item) => (
      <Item key={item._id} {...item} />
    ))}
  </ul>
)

const ItemsConnected = () => {
  const { data } = useItems() // custom hook
  return <Items items={data} />
}
```

All the items are actually re-rendered even if **the server response is exactly the same**. How could I setup react-query to re-render **only** if the data actually changed ?

**Comment by TkDodo** [comment, marked as answer]
it's re-rendering because of the `isFetching` transition. Please read. https://tkdodo.eu/blog/react-query-render-optimizations

tracked queries solve this. that being said, please read the disclaimer. rendering a list of items shouldn't be a problem, even if you do it  "unnecessarily" from time to time :)

**Reply by mlisthenewcool** [reply]
Works like a charm, thanks for the quick answer !
